src/meta/ActiveHostsMan.cpp

```diff
@@ -120,6 +120,26 @@ void ActiveHostsMan::cleanExpiredHosts() {
         }
     }
 
+    // merge host info from kvstore
+    if (kvstore_ != nullptr) {
+        const auto& prefix = MetaServiceUtils::hostPrefix();
+        std::unique_ptr<kvstore::KVIterator> iter;
+        auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+        if (ret == kvstore::ResultCode::SUCCEEDED) {
+            while (iter->valid()) {
+                auto host = MetaServiceUtils::parseHostKey(iter->key());
+                if (iter->val() == MetaServiceUtils::hostValOnline()) {
+                    bool found = hostsMap_.find({host.ip, host.port}) == hostsMap_.end();
```

Comment:
Maybe you need a lock here

Reply:
Also maybe just use if (hostsMap_.find({host.ip, host.port}) == hostsMap_.end()) here, you name it "found", which means "not found in memory", it confuses.

Reply:
Here is another case, what if it is offline in kvstore, but online in memory, should we handle it? If we don't, those host will stay offline forever, the result of getActiveHosts and "show hosts" will be different.